src/avram/tasks/db/create.cr

```diff
@@ -19,8 +19,37 @@ class Db::Create < LuckyCli::Task
   end
 
   def call
-    Avram::Migrator.run do
-      Avram::Migrator::Runner.create_db(@quiet)
+    result, output = run_connection_check
+    if result.exit_code == 0
+      Avram::Migrator.run do
+        Avram::Migrator::Runner.create_db(@quiet)
+      end
+    else
+      uri = URI.parse(connection_url)
+      raise Avram::ConnectionError.new(uri, Avram.settings.database_to_migrate)
     end
   end
+
+  private def connection_url : String
+    Avram::PostgresURL.build(
+      database: "",
+      hostname: Avram::Migrator::Runner.db_host.to_s,
+      username: Avram::Migrator::Runner.db_user.to_s,
+      password: Avram::Migrator::Runner.db_password.to_s,
+      port: Avram::Migrator::Runner.db_port.to_s
+    )
+  end
+
+  private def run_connection_check
+    output = IO::Memory.new
+    command = "psql -q -l #{connection_url}"
```

Comment:
Is there a way to do this through the postgres wire protocol?

The only reason I ask is because I dont know if this will be a command available on boxes that only install the postgres client libraries.  It might be a CLI util distributed with any postrges package and in that case it is less of an issue. 

Also it feels weird to subshell out every time but create is run once so I dont think that is a big deal.

Reply:
Not that I've found. There's tools like `pg_isready`, and a few other options, but all of them require you have the postgres client tools installed which you need to call `createdb` anyway, so you'll be guaranteed to have `psql` installed. 

We actually shell out for a lot of different commands [createdb](https://github.com/luckyframework/avram/blob/master/src/avram/migrator/runner.cr#L64), [dropdb](https://github.com/luckyframework/avram/blob/master/src/avram/migrator/runner.cr#L54), [restore](https://github.com/luckyframework/avram/blob/master/src/avram/migrator/runner.cr#L107), and [dump](https://github.com/luckyframework/avram/blob/master/src/avram/migrator/runner.cr#L118). 

If the `pg` shard ends up supporting all of these natively through a C API where we don't have to even have these CLI tools installed, I'd love to move to that. Probably a way down the road update though.

Reply:
sounds great I was just asking :)

Reply:
we actually might be able to reduce a lot of these calls.  There is direct SQL for them https://www.postgresql.org/docs/9.1/manage-ag-createdb.html

Reply:
That's true. It would be good to start looking in to how we can clean all this up. After this next release I definitely want to focus on some "spring" cleaning of all the repos. Thanks for the heads up!